src/components/tooltip/tooltip.test.tsx

```diff
@@ -0,0 +1,66 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import { userEvent } from '@testing-library/user-event';
+import { Tooltip, type ITooltipProps } from './tooltip';
+
+describe('<Tooltip/> component', () => {
+    const originalResizeObserver = global.ResizeObserver;
+
+    beforeAll(() => {
+        // mocking ResizeObserver since it's not available in Jest test environment
+        global.ResizeObserver = jest.fn().mockImplementation(() => ({
+            observe: jest.fn(),
+            unobserve: jest.fn(),
+            disconnect: jest.fn(),
+        }));
+    });
+
+    afterAll(() => {
+        global.ResizeObserver = originalResizeObserver;
+    });
```

Comment:
I would suggest to use `beforeEach` instead so that tests don't depend on the same mocked instance
```suggestion
    beforeEach(() => {
        // mocking ResizeObserver since it's not available in Jest test environment
        global.ResizeObserver = jest.fn().mockImplementation(() => ({
            observe: jest.fn(),
            unobserve: jest.fn(),
            disconnect: jest.fn(),
        }));
    });

    afterEach(() => {
        global.ResizeObserver = originalResizeObserver;
    });
```